addons/account_asset/models/account_asset.py

```diff
@@ -103,6 +112,21 @@ class AccountAssetAsset(models.Model):
         help="It is the amount you plan to have that you cannot depreciate.")
     invoice_id = fields.Many2one('account.invoice', string='Invoice', states={'draft': [('readonly', False)]}, copy=False)
     type = fields.Selection(related="category_id.type", string='Type', required=True)
+    date_first_depreciation = fields.Selection([
+        ('first_day_month', 'Purchase Date'),
```

Comment:
ok then I propose to do
('purchase_date', 'Purchase Month') and help: If set as 'Purchase Month', it will take the first day of the month the supplier bill belong to.